test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -126,6 +128,26 @@ protected void afterAlways(List<Throwable> errors) throws Throwable {
         }
     });
 
+    /**
+     * helper to randomly perform on <code>consumer</code> with <code>value</code>
+     */
+    public static <T> void maybeSet(Consumer<T> consumer, T value) {
```

Comment:
nit: can both new helper methods go somewhere further down in ESTestCase? From the comments it seems someone tries to keep some order here, and this would be the setup / cleanup section. Also maybe both should be public? Although protected also works, we need it in tests that subclass this anyway.